### PR TITLE
Bump spring to 2.3.0

### PR DESCRIPTION
**Pull request by Wirick**
1. This induces upgrades for a number of different packages in order to be
compatible, including kafka and the jakarta validation-api
2. Resolved logging errors in core/serving from seeing different logger
   classes in the classpath by excluding them

I am able to build and run feast core and feast serving, along with bigquery batch ingestion with these updated dependencies. However there was an upgrade needed in how we mock the url for the test database. Context: https://stackoverflow.com/questions/49088847/after-spring-boot-2-0-migration-jdbcurl-is-required-with-driverclassname

### CODE REVIEW

**Comment by ches** [comment]
The re-indentation is a bit distracting, I know some POM files in the project are indented with two spaces and others four, and GitHub has the option to ignore whitespace in diffs, but if we're going to do mass reformats it'd be nice to do them in separate commits and ideally with a means to check/automate the convention going forward.

Something that could have been brought up in #793 I think: we use an imported Spring bill of materials from the root `pom.xml` which sets managed versions of dependencies for Spring ecosystem—it'd be preferable IMO to let that drive versions of things like spring-security deps (i.e. just drop the `<version>` tags). Unless we have a specific reason for deviating from the BOM, which should be documented with a comment.

Spring and Beam or proto/gRPC eco dep versions may be at odds sometimes so may necessitate overrides, but at least as long as Core depends on Ingestion I believe it's worth maintaining the BOM usage. If/when that situation changes it might make sense to move service API modules under their own parent POM holding the Spring BOM.

File: pom.xml [hunk 1]
```diff
@@ -366,6 +374,9 @@
                         <goals>
                             <goal>jar</goal>
                         </goals>
+                        <configuration>
+                            <javadocExecutable>${java.home}/bin/javadoc</javadocExecutable>
```
Comment (ches) [inline pom.xml:378]: Any reason this needed to set specifically?

**Comment by Wirick** [comment]
Closing in favor of https://github.com/feast-dev/feast/pull/854

File: pom.xml (hunk 1 above)
Comment (Wirick) [reply pom.xml:378]: Yea my intellij environment ran into the issue described and solved here https://intellij-support.jetbrains.com/hc/en-us/community/posts/206880855-MavenReportException-Error-while-creating-archive-Unable-to-find-javadoc-command-The-environment-variable-JAVA-HOME-is-not-correctly-set- so thats what allowed me to execute this in intellij. We can continue the convo here https://github.com/feast-dev/feast/pull/854

**Pull request closed without merging**